nextcord/ext/commands/bot.py

```diff
@@ -639,8 +644,10 @@ def add_cog(self, cog: Cog, *, override: bool = False) -> None:
                 raise nextcord.ClientException(f"Cog named {cog_name!r} already loaded")
             self.remove_cog(cog_name)
 
-        cog = cog._inject(self)
-        self.__cogs[cog_name] = cog
+        if isinstance(cog, Cog):
+            cog = cog._inject(self)
+            self.__cogs[cog_name] = cog
```

Comment:
This probably can still happen since you already allowed __cogs to contain nextcord.Cog types and it's already using it to check if it already exists.

Maybe some of this can be moved to nextcord.Client so it can check for duplicates and consolidate `Bot.__cogs` and `Client._client_cogs`?

```suggestion
        self.__cogs[cog_name] = cog
```